tap_github/streams.py

```diff
@@ -375,3 +375,112 @@ def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
             ),
         ),
     ).to_dict()
+
+
+class CommitsStream(GitHubStream):
+    """
+    Defines the 'Commits' stream.
+    The stream is fetched per repository to maximize optimize for API quota
+    usage.
+    """
+
+    name = "commits"
+    path = "/repos/{org}/{repo}/commits"
+    primary_keys = ["node_id"]
+    replication_key = "commit_timestamp"
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys = ["repo", "org"]
+    ignore_parent_replication_key = True
+
+    def get_url_params(
+        self, context: Optional[dict], next_page_token: Optional[Any]
+    ) -> Dict[str, Any]:
+        """Return a dictionary of values to be used in URL parameterization."""
+        params = super().get_url_params(context, next_page_token)
+        if self.replication_key:
+            since = self.get_starting_timestamp(context)
+            if since:
+                params["since"] = since
+        return params
+
+    def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
+        """
+        Add a timestamp top-level field to be used as state replication key.
+        It's not clear from github's API docs which time (author or committer)
+        is used to compare to the `since` argument that the endpoint supports.
+        """
+        row["commit_timestamp"] = row["commit"]["committer"]["date"]
+        return row
+
+    schema = th.PropertiesList(
+        th.Property("node_id", th.StringType),
+        th.Property("url", th.StringType),
+        th.Property("sha", th.StringType),
+        th.Property("html_url", th.StringType),
+        th.Property("commit_timestamp", th.DateTimeType),
+        th.Property(
+            "commit",
+            th.ObjectType(
+                th.Property(
+                    "author",
+                    th.ObjectType(
+                        th.Property("name", th.StringType),
+                        th.Property("email", th.StringType),
+                        th.Property("date", th.DateTimeType),
+                    ),
+                ),
+                th.Property(
+                    "committer",
+                    th.ObjectType(
+                        th.Property("name", th.StringType),
+                        th.Property("email", th.StringType),
+                        th.Property("date", th.DateTimeType),
+                    ),
+                ),
+                th.Property("message", th.StringType),
+                th.Property(
+                    "tree",
+                    th.ObjectType(
+                        th.Property("url", th.StringType),
+                        th.Property("sha", th.StringType),
+                    ),
+                ),
+                th.Property("comment_count", th.IntegerType),
```

Comment:
I think we should remove `comment_count` since we want to treat commits as immutable and pull them incrementally based on commit_timestamp. 

```suggestion
```

Reply:
A separate stream of commit comments could add back this metric based on comment timestamp.

Reply:
Yes, this makes sense! I guess I will just add the `commit_comments` stream as well :)

Reply:
@laurentS - do you mind removing `commit_comments` from properties here? Since this will be desynced from the source (due to `commits` being immutable and the count not incrementing the replication key `commit_timestamp`), I think we should keep this out of this particular stream and allow users to calculate in another way.